src/keycloak_utils/consumer/rest_framework/core.py

```diff
@@ -0,0 +1,288 @@
+
+        def _register_queues_from_dict(self, queue_dict, queue_status):
+            for queue_type, queues in queue_dict.items():
+                for queue_name in queues:
+                    routing_key = (
+                        f"eventbus.{queue_type}.{queue_name}"
+                        if queue_type != "general"
+                        else "#"
+                    )
+                    queue_name = (
+                        f"{queue_type}.{queue_name}"
+                        if queue_type != "general"
+                        else queue_name
+                    )
+                    self._registry[queue_status].append(
+                        {"queue": queue_name, "routing_key": routing_key}
+                    )
+
+        def register_queue(self, queue, routing_key=None, queue_status="create"):
+            if queue_status not in self._registry:
+                raise ValueError(f"Unsupported queue status: {queue_status}")
+
+            if isinstance(queue, dict):
+                self._register_queues_from_dict(queue, queue_status)
+            else:
+                self._registry[queue_status].append(
+                    {"queue": queue, "routing_key": routing_key}
+                )
+
+        def get_registry(self):
+            """Retrieve the complete registry."""
+            return self._registry
+
+        def get_queues(self, queue_type):
+            return self._registry.get(queue_type, None)
+
+        def list_queues_dict(self):
+            return self._registry
+
+    def on_queue_declared(self, method_frame):
+        queue_name = method_frame.method.queue
+        logger.info(f"consuming Queue {queue_name}")
+        exception_map = {
+            AMQPConnectionError: "Connection error",
+            AMQPChannelError: "Channel error",
+            ConnectionClosedByBroker: "Connection closed by broker",
+        }
+        try:
+            self.channel.basic_consume(
+                queue=queue_name, on_message_callback=self.handle_message
+            )
+        except Exception as e:
+            message = exception_map.get(type(e), "Unexpected error")
+            logger.error(f"{message}: {e}")
+            self.retry_connect()
+
+    def stop(self):
+        self.should_stop.set()
+        if self.channel and self.channel.is_open:
+            self.connection.close()
+        logger.info("Stopped consuming messages")
+
+    def reject_callback(self, ch, method, properties, body):
+        logger.info(f"Rejecting message: {self.decode_event(body)}")
+        ch.basic_reject(delivery_tag=method.delivery_tag, requeue=False)
+
+    @staticmethod
+    def decode_event(body):
+        return msgpack.unpackb(body, raw=False)
+
+    def handle_message(self, ch, method, properties, body):
+        try:
+            event_data = self.decode_event(body)
+            self.process_message(event_data)
+            ch.basic_ack(delivery_tag=method.delivery_tag)
+        except Exception as e:
```

Comment:
same comment for Exceptions, we need to know which exceptions can our code trigger and handle them accordingly.

Reply:
Just saw down the line you are using KeyboardIntrerupt. you know that `Exception` is also suppressing that one also, will have no effect?